packages/flame_riverpod/example/lib/main.dart

```diff
@@ -0,0 +1,158 @@
+    StateNotifierProvider<RiverpodAwareGameNotifier, FlameGame?>((ref) {
+  return RiverpodAwareGameNotifier();
+});
+
+/// Simple [StateNotifier] that holds the current [FlameGame] instance.
+class RiverpodAwareGameNotifier extends StateNotifier<FlameGame?> {
+  RiverpodAwareGameNotifier() : super(null);
+
+  void set(FlameGame candidate) {
+    state = candidate;
+  }
+}
+
+void main() {
+  runApp(const ProviderScope(child: MyApp()));
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      home: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: const [
+          Expanded(child: FlutterCountingComponent()),
+          Expanded(
+            child: RiverpodGameWidget.initializeWithGame(
+              uninitializedGame: RefExampleGame.new,
+            ),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class FlutterCountingComponent extends ConsumerWidget {
+  const FlutterCountingComponent({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final textStyle = Theme.of(context)
+        .textTheme
+        .headlineSmall
+        ?.copyWith(color: Colors.white);
+
+    final stream = ref.watch(countingStreamProvider);
+    return Material(
+      color: Colors.transparent,
+      child: Column(
+        children: [
+          Text('Flutter', style: textStyle),
+          stream.when(
+            data: (value) => Text('$value', style: textStyle),
+            error: (error, stackTrace) => Text('$error', style: textStyle),
+            loading: () => Text('Loading...', style: textStyle),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class RefExampleGame extends FlameGame with HasComponentRef {
+  RefExampleGame(WidgetRef ref) {
+    // Note: WidgetRef is not stored and is used to build a [ComponentRef],
+    // a wrapper around it.
+    HasComponentRef.widgetRef = ref;
+  }
+
+  @override
+  Future<void> onLoad() async {
+    await super.onLoad();
+    add(TextComponent(text: 'Flame'));
+    add(RiverpodAwareTextComponent());
+  }
+}
+
+class RiverpodGameWidget extends ConsumerStatefulWidget {
+  const RiverpodGameWidget.readFromProvider({super.key})
+      : uninitializedGame = null;
+  const RiverpodGameWidget.initializeWithGame({
+    super.key,
+    required this.uninitializedGame,
+  });
+
+  final FlameGame Function(WidgetRef ref)? uninitializedGame;
+
+  @override
+  ConsumerState<RiverpodGameWidget> createState() => _RiverpodGameWidgetState();
+}
+
+class _RiverpodGameWidgetState extends ConsumerState<RiverpodGameWidget> {
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      if (widget.uninitializedGame is FlameGame Function(WidgetRef ref)) {
+        ref
+            .read(riverpodAwareGameProvider.notifier)
+            .set(widget.uninitializedGame!(ref));
```

Comment:
Any chance this logic should be moved to a provider instead of in a widget?
To remove the addPostFrameCallback which will likely be problematic

Like:

```dart
final gameProvider = Provider((ref) => RefExampleGame());
```

Reply:
In the initial version of the package I did make a decision along those lines (among others, e.g. including a widget that extended ConsumerStatefulWidget as part of the package), but I became convinced that it may be better left to application developers to define these. I'm not especially committed to one position or another

Reply:
I'm not sure we're talking about the same thing. I'm not talking about having such a provider in the package itself

I meant in this example specifically. I don't think a StateNotifierProvider is needed here